### ISSUE TITLE
Provide examples

### ISSUE THREAD

**Issue by jdistlr** [etmendz/Mendz.ETL#1]
This looks kind of mighty! Will you provide examples?

**Comment by etmendz** [comment]
Hi @jdistlr! Thank you for your interest in Mendz.ETL. So far, I only have a guide on the basic steps that you can take to implement an ETL app/solution at https://github.com/etmendz/Mendz.ETL/wiki. Note that I have documented the essential ingredients in the wiki. The "examples" and "remarks" in the wiki provide ideas on how you can make it work. Let me know how far you are in following them and maybe I can help if you are struggling on something.

The reason I don't have concrete examples is because of the invitation to invent, thus letting developers take ownership of their implementations. If I do provide any in the future, it would rather be full projects/products like Mendz.ETL.JSON, Mendz.ETL.Xml, Mendz.ETL.Csv, Mendz.ETL.Text, etc. for example. Still, the innards of these libraries would show that they follow the basic steps provided in https://github.com/etmendz/Mendz.ETL/wiki.

**Comment by etmendz** [comment]
I have more articles about Mendz.ETL here: https://asiseetech.blogspot.com/search?q=mendz.etl. Hope these help. I am open to creating concrete examples, but if you can't wait, so far I have these articles to provide ideas on how to start creating your own ETL app/solution.

**Comment by jjwilliams42** [comment]
I have to agree. Your library sounds fantastic! But it's missing the key ingredient for adoption: any actual examples of using it.

**Comment by etmendz** [comment]
I hear you. I will work on it. For now, please look at Mendz.ETL.Common, where you'll find classes that can give you basic ideas on how to create adapters and mappers. If you'll try to put together the LineSourceAdapter, PassThroughMapper and LineTargetAdapter in an application, the code can look like the ff.:

```C#
// Initialize document specifications
DocumentSpecification sourceSpecification = new DocumentSpecification { Address = @"C:\Temp\source.txt" };
DocumentSpecification targetSpecification = new DocumentSpecification { Address = @"C:\Temp\target.txt" };
// Initialize the source adapter
ISourceAdapter source = new LineSourceAdapter();
source.SourceSpecification = sourceSpecification;
// Initialize the mapper
IMapper mapper = new PassThroughMapper();
// Initialize the target adapter
ITargetAdapter target = new LineTargetAdapter();
target.TargetSpecification = targetSpecification;
// Pass everything to the Router...
Router.Route(source, mapper, target);
```

The use of PassThroughMapper above results to simply copying the contents of source.txt to target.txt. If you create your own mapper, you can implement TransformInputToOutput() to make changes to the input before returning it. Doing so saves your mapper's result to target.txt instead. Thus, by simply changing the mapper, you can make the same program construct above return different results.

Hope this helps... for now.
